taxonomy-editor-frontend/src/pages/editentry/ListTranslations.jsx

```diff
@@ -16,17 +17,24 @@ const ListTranslations = ({ nodeObject, setNodeObject }) => {
 
     const [renderedTranslations, setRenderedTranslations] = useState([]) // Stores state of all tags
     const [mainLangRenderedTranslations, setMainLangRenderedTranslations] = useState([]) // Stores state of main language's tags
-    const [openDialog, setOpen] = useState(false); // Used for Dialog component
+    const [openDialog, setOpenDialog] = useState(false); // Used for Dialog component
     const [newLanguageCode, setNewLanguageCode] = useState(''); // Used for storing new LC from Dialog
-    const [isValidLanguageCode, setisValidLanguageCode] = useState(false); // Used for validating a new LC
+    const [languageAction, setLanguageAction] = useState('invalid'); // Used for storing state of action to be performed on LC (add/show/invalid)
+    const [shownTranslationLanguages, setShownTranslationLanguages] = useState([]); // Used for storing LC's that are to be shown
+    const [addedLanguageCodes, setAddedLanguageCodes] = useState([]); // Used for storing LC's that are added to the nodeObject
+    const translationVisibilityPreferenceKey = "shownTranslationLanguages" // Used for storing LC's that are to be shown in localStorage
```

Comment:
this one can live outside of the component (like before `const ListTranslations = ({ nodeObject, setNodeObject }) => {`). Please update comment, to something like: `// local storage key for languages to be shown`